R/package.R

```diff
@@ -227,9 +227,41 @@ do_check <- function(
   }
 
   # user supplied env vars take precedence
-  if (length(env)) chkenv[names(env)] <- env
+  if (length(env)) {
+    chkenv[names(env)] <- env
+  }
+
+  if (!quiet) {
+    cat_head("R CMD check")
+    cat_line()
+    all_vars <- Sys.getenv()
+    rchk_vars <- all_vars[
+      grep("_R_CHECK|NOT_CRAN|RCMDCHECK|R_TESTS", names(all_vars))
+    ]
+    rchk_vars[names(env)] <- env
```

Comment:
In this context, I think it's ok to just not print anything if there are no env vars set (since that's going to be rare)